src/averageAges.js

```diff
@@ -36,8 +45,22 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = !!withChildren
+    ? people.filter(person =>
+      person.sex === 'f' && people.some(child => child.mother === person.name))
+    : people.filter(person => person.sex === FEMALE);
+
+  const womenTotalAge = women.reduce((acc, woman) => {
+    return acc + (woman.died - woman.born);
+  }, 0);
+
+  womenTotalAge.sort().reverse();
+
+  const averafeWoomanAge = womenTotalAge / women.length + 1;
+
+  return averafeWoomanAge;
 }
```

Comment:
The function 'calculateWomenAverageAge' has a typo in the variable names. 'womenTotalAge' should be an array to store individual ages, not the total age. Also, there is a sorting function on line 59 that is unnecessary and incorrect. The calculation of average age is incorrect as well.